components/gitpod-cli/cmd/git-token-validator.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+
+	serverapi "github.com/gitpod-io/gitpod/gitpod-protocol"
+	supervisor "github.com/gitpod-io/gitpod/supervisor/api"
+)
+
+var (
+	user        string
+	token       string
+	tokenScopes string
+	host        string
+	repoURL     string
+	gitCommand  string
+)
```

Comment:
In some code we've started to wrap those global flag variables into a struct to not pollute the package so much, e.g.
```suggestion
var gitTokenValidatorOpts struct {
	User        string
	Token       string
	TokenScopes string
	Host        string
	RepoURL     string
	GitCommand  string
}
```

Reply:
Good idea. I actually ran into a problem once with these package global vars.